src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -9,36 +9,43 @@
 import java.util.stream.Collectors;
 
 public class StreamStudy {
-
     public static long countWords() throws IOException {
         String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+                                                            .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-
-        long count = 0;
-        for (String w : words) {
-            if (w.length() > 12) count++;
-        }
-        return count;
+        return words.stream()
+                    .filter(word -> word.length() > 12)
+                    .count();
     }
 
     public static void printLongestWordTop100() throws IOException {
         String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
+                                                            .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-
-        // TODO 이 부분에 구현한다.
+        words.stream()
+             .filter(word -> word.length() > 12)
+             .distinct()
+             .sorted((p1, p2) -> Integer.compare(p2.length(), p1.length()))
+             .limit(100)
+             .map(String::toLowerCase)
+             .forEach(System.out::println);
```

Comment:
```suggestion
            .map(String::toLowerCase)
            .collect(Collectors.joining("\n"));
        System.out.println(joinedWords);
```
표준 출력을 사용할 일이 거의 없겠지만
System.out.println 같은 I/O 작업을 100건 호출하는 방식보다는
문자열을 만들고 한번에 출력하는 것이 더 좋을 것 같아요

Reply:
넵! 감사합니다! 반영하겠습니다